### PR TITLE
Backport the fix for the issue #3136 from strimzi 0.21 to 0.19

### PR DESCRIPTION
**Pull request by yyang48**
Issue: https://github.com/strimzi/strimzi-kafka-operator/issues/3136

Original fix: https://github.com/strimzi/strimzi-kafka-operator/pull/4153

Signed-off-by: Yaodong Yang <yyang48@apple.com>

### Type of change

_Select the type of your PR_

- Bugfix
- Enhancement / new feature
- Refactoring
- Documentation

### Description

_Please describe your pull request_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [x] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java [hunk 1]
```diff
@@ -53,6 +35,24 @@
...
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Function;
+import java.util.function.Supplier;
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java:54]: Maybe the re-ordering here can be avoided?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaRollerTest.java [hunk 2]
```diff
@@ -44,6 +32,18 @@
...
+import java.util.Collection;
+import java.util.Collections;
+import java.util.IdentityHashMap;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.stream.Collectors;
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaRollerTest.java:45]: Same here.

**Review by scholzj — COMMENTED** [review]
The code looks good in general. But it looks like the CI pipeline does nto work anymore after almost 2 years. So I do not think there is an easy way to test this in the CI. The error seems to suggest that you would need to hardcode some older version of Minikube into the https://github.com/strimzi/strimzi-kafka-operator/blob/release-0.19.x/.travis/setup-kubernetes.sh script. Bu not sure if that will be the only issue.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java (hunk 1 above)
Comment (yyang48) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java]: done

**Pull request closed without merging**